bin/wsl/diag.wsl2.sh

```
@@ -0,0 +1,391 @@
+# ref1: https://regex101.com/r/uwPxJf/10
+function color:ipv6() {
+    local line=$1 grep_regex seeds
+    local ipv6s ipv4s seeds
+
+    # Combine all IPv6 patterns
+    grep_regex=$(ipv6:grep)
+    local sed_regex="${grep_regex}"
+
+    # Extract IPv6 addresses
+    ipv6s=$(echo "$line" | grep -oPI "${grep_regex}")
+    ipv4s=$(echo "$line" | grep -oPI '([0-9]{1,3}\.){3}[0-9]{1,3}') # IPv4 addresses format
```

Shellcheck (0.10.0)

[warning] ipv4s appears unused. Verify use (or export if used externally).

(SC2034)